src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -216,10 +208,10 @@ else if (shouldCheckLastRcurly) {
                 violation = MSG_KEY_LINE_ALONE;
             }
         }
-        else if (shouldBeOnSameLine(bracePolicy, details)) {
+        else if (shouldBeOnSameLine(option, details)) {
             violation = MSG_KEY_LINE_SAME;
         }
-        else if (shouldBeAloneOnLine(bracePolicy, details)) {
+        else if (shouldBeAloneOnLine(option, details)) {
             violation = MSG_KEY_LINE_ALONE;
         }
         else if (shouldStartLine && !isOnStartOfLine(details, targetSourceLine)) {
```

Comment:
I would make this only `else if (shouldStartLine) {` and move the variable declaration of `targetSourceLine` to the next line.